crates/sncast/src/starknet_commands/deploy.rs

```diff
@@ -46,19 +57,113 @@ pub struct Deploy {
     pub version: Option<DeployVersion>,
 }
 
+impl Deploy {
+    pub fn build_artifacts(
+        &self,
+        json: bool,
+        profile: &Option<String>,
+    ) -> Result<CompiledContract> {
+        let contract_name = self
+            .contract_name
+            .clone()
+            .ok_or(anyhow!("Contract name unspecified"))?;
+
+        let artifacts = read_manifest_and_build_artifacts(&self.package, json, profile)?;
+
+        let contract_artifacts = artifacts.get(&contract_name).ok_or(anyhow!(
+            "No artifacts found for contract: {}",
+            contract_name
+        ))?;
+
+        CompiledContract::from(contract_artifacts)
+    }
+
+    pub fn declare_data(&self) -> Result<Declare> {
+        if self.contract_name.is_none() {
+            bail!("Contract name unspecified");
+        }
+
+        let Deploy {
+            contract_name,
+            package,
+            fee_args,
```

Comment:
This copy-based behavior on fees is worrying for me, since the user might not be aware that it implies the fee might be paid TWICE since both the declaration and deployment will consume the specified fee. 

Let's discuss about potential solutions to this problem offline